modules/ocl/cl_fisheye_handler.cpp

```diff
@@ -374,44 +375,72 @@ CLFisheyeHandler::generate_fisheye_table (
         ERROR, _geo_table.ptr () && _geo_table->is_valid (),
         XCAM_RETURN_ERROR_MEM, "[%s] check geo map buffer failed", get_name ());
 
-    CLArgList args;
-    CLWorkSize work_size;
-
-    FisheyeInfo fisheye_arg1 = fisheye_info;
-    fisheye_arg1.wide_angle = degree2radian (fisheye_info.wide_angle);
-    fisheye_arg1.rotate_angle = degree2radian (fisheye_info.rotate_angle);
-    args.push_back (new CLArgumentT<FisheyeInfo> (fisheye_arg1));
-
-    float fisheye_image_size[2];
-    fisheye_image_size[0] = fisheye_width;
-    fisheye_image_size[1] = fisheye_height;
-    args.push_back (new CLArgumentTArray<float, 2> (fisheye_image_size));
-    args.push_back (new CLMemArgument (_geo_table));
-
-    float radian_per_pixel[2];
-    radian_per_pixel[0] = degree2radian (longitude / table_width);
-    radian_per_pixel[1] = degree2radian (latitude / table_height);
-    args.push_back (new CLArgumentTArray<float, 2> (radian_per_pixel));
-
-    float table_center[2];
-    table_center[0] = table_width / 2.0f;
-    table_center[1] = table_height / 2.0f;
-    args.push_back (new CLArgumentTArray<float, 2> (table_center));
-
-    work_size.dim = 2;
-    work_size.local[0] = 8;
-    work_size.local[1] = 4;
-    work_size.global[0] = XCAM_ALIGN_UP (table_width, work_size.local[0]);
-    work_size.global[1] = XCAM_ALIGN_UP (table_height, work_size.local[1]);
-
-    XCAM_FAIL_RETURN (
-        ERROR, table_kernel->set_arguments (args, work_size) == XCAM_RETURN_NO_ERROR,
-        XCAM_RETURN_ERROR_CL, "kernel_fisheye_table set arguments failed");
-
-    XCAM_FAIL_RETURN (
-        ERROR, table_kernel->execute (table_kernel, true) == XCAM_RETURN_NO_ERROR,
-        XCAM_RETURN_ERROR_CL, "[%s] execute kernel_fisheye_table failed", get_name ());
-
+    if(_surround_mode == BowlView) {
+        BowlDataConfig bowl_data_config = get_bowl_config();
```

Comment:
too many lines there.
it's better change to 

    if (bowl_view)
           fill_bowl_view_table ();
    else
            fill_sphere_view_table ();